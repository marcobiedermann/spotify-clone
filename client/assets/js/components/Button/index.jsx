import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.css';

const Button = ({ children }) => (
  <button
    className={styles.button}
    type="button"
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.node,
};

Button.defaultProps = {
  children: null,
};

export default Button;
