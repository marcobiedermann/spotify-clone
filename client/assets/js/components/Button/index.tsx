import classNames from 'classnames';
import React, { FC } from 'react';
import styles from './style.module.css';

export interface ButtonProps {
  className?: string;
  type?: 'button' | 'reset' | 'submit';
}

const Button: FC<ButtonProps> = (props) => {
  const { className, type = 'button', ...otherProps } = props;

  return <button className={classNames(className, styles.button)} type={type} {...otherProps} />;
};

export default Button;
