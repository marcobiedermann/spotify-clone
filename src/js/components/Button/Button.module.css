.button {
  background-color: #38bc54;
  border-radius: 999em;
  color: #fff;
  display: inline-block;
  min-width: 10em;
  padding: 0.75em 1.5em;
  text-transform: uppercase;
  transition: transform 0.3s;
}

.button:focus,
.button:hover {
  transform: scale(1.05);
}
