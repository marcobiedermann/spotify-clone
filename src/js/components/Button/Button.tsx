import clsx from 'clsx';
import { ReactNode } from 'react';
import styles from './Button.module.css';

export interface ButtonProps {
  className?: string;
  type?: 'button' | 'reset' | 'submit';
  children: ReactNode;
}

function Button(props: ButtonProps): JSX.Element {
  const { className, type = 'button', ...otherProps } = props;

  return <button className={clsx(className, styles.button)} type={type} {...otherProps} />;
}

export default Button;
