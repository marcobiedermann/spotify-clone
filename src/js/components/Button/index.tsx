/* eslint-disable react/button-has-type */

import classNames from 'classnames';
import React, { ReactNode } from 'react';
import styles from './style.module.css';

export interface ButtonProps {
  className?: string;
  type?: 'button' | 'reset' | 'submit';
  children: ReactNode;
}

function Button(props: ButtonProps): JSX.Element {
  const { className, type = 'button', ...otherProps } = props;

  return <button className={classNames(className, styles.button)} type={type} {...otherProps} />;
}

export default Button;
