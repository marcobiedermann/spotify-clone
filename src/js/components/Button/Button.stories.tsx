import type { Meta, StoryObj } from '@storybook/react';
import Button from '.';

const meta: Meta<typeof Button> = {
  component: Button,
  args: {
    children: 'Button',
    type: 'button',
  },
};

type Story = StoryObj<typeof Button>;

const Default: Story = {};

export { Default };
export default meta;
