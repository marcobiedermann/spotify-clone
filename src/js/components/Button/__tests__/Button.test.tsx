import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';
import Button from '..';

it('renders correctly', () => {
  const { container } = render(<Button>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});
