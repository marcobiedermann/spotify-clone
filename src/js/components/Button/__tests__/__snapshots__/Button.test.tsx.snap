// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<button
  class="button"
  type="button"
>
  Button
</button>
`;
